<?xml version="1.0"?>
<launch>
  <arg name="uri" default="radio://0/80/2M/E7E7E7E7E7" /> <!--Adjust this parameter-->
  <arg name="frame" default="crazyflie1" /> <!--For OptiTrack users: frame name should match the Rigid-body name on Motive-->
  <arg name="ip" default="172.16.1.1" /> <!--Adjust this parameter-->
  <arg name="port" default="3883" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="crazyflie1">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri)" />
      <param name="tf_prefix" value="crazyflie1" />
      <param name="enable_logging" value="False" /> <!--Logging is disabled to reduce unnecessary bandwith consumption-->
      <param name="enable_logging_imu" value="False" />
      <param name="enable_logging_temperature" value="False" />
      <param name="enable_logging_magnetic_field" value="False" />
      <param name="enable_logging_pressure" value="False" />
      <param name="enable_logging_battery" value="False" />
      <param name="enable_logging_packets" value="False" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["stateEstimate.x", "stateEstimate.y", "stateEstimate.z"]
      </rosparam>
    </node>

    <node name="pose" pkg="crazyflie_demo" type="publish_external_position_vrpn.py" output="screen"> 
      <param name="topic" value="/crazyflie1/vrpn_client_node/crazyflie1/pose" />
    </node>

    <!--Run vrpn client-->
    <!--For OptiTrack users: make sure you edit vrpn_client_ros.cpp file to adjust axes transformation-->
    <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      server: $(arg ip)
      port: $(arg port)
      update_frequency: 100.0
      frame_id: /world
      child_frame_id: $(arg frame)
      use_server_time: false
      broadcast_tf: false
      refresh_tracker_frequency: 1.0
    </rosparam>
    </node>
  </group>
</launch>
